### PR TITLE
doc: Bluetooth: Mesh: Docs automated DFU meta gen

### PR DESCRIPTION
**Pull request by Andrewpini**
Add documentation for automated DFU metadata generation.

### CODE REVIEW

**Comment by Andrewpini** [comment]
This PR is blocked by https://github.com/nrfconnect/sdk-nrf/pull/14074. Will rebase when it is merged.

File: doc/nrf/protocols/bt/bt_mesh/dfu_over_bt_mesh.rst [hunk 1]
```diff
@@ -182,8 +182,31 @@ To bring a stalled Target node to idle state, use the ``mesh models dfu cli canc
...
+   the metadata string(s). Additionally, the script is hardcoded to produce a metadata string where the firmware is targeted for the application core.
+
+In addition to printing the metadata string on build, a separate west command can be utilized to print the metadata to the console, given that it is already generated by the build system. This provides the user
+easy access to this information without having to enter the .json file in the build folder or to rebuild the application::
+
+  west ble-mesh-dfu-metadata
+
+For this particular example, the output generated by either the build system or the west ble-mesh-dfu-metadata command would be::
+
+  Encoded metadata: 020000000100000094cf24017c26f3710100
```
Comment (PavelVPV) [inline doc/nrf/protocols/bt/bt_mesh/dfu_over_bt_mesh.rst:206]: The complete script output should be added here:
https://github.com/nrfconnect/sdk-nrf/blob/7d209a57228062b2453d8ba25b172e8d09a26d49/scripts/bluetooth/mesh/mesh_dfu_metadata.py#L408-L410

File: doc/nrf/protocols/bt/bt_mesh/dfu_over_bt_mesh.rst [hunk 2]
```diff
@@ -182,8 +182,31 @@ To bring a stalled Target node to idle state, use the ``mesh models dfu cli canc
...
-Composing the firmware metadata
-*******************************
+Generating the firmware metadata
+********************************
+
+Automated metadata generation
+=============================
+
+The most efficient way to generate the required DFU metadata is to use the DFU metadata extraction script integrated in the NCS build system. By enabling the ``CONFIG_BT_MESH_DFU_METADATA_ON_BUILD`` option in the
+application, the metadata will be automatically parsed from the .elf and .config file. The parsed data is put into a DFU zip-archive in the build folder of the application in a JSON format. Additionally, the
```
Comment (PavelVPV) [inline doc/nrf/protocols/bt/bt_mesh/dfu_over_bt_mesh.rst:192]: We can say the archive and file name. Other FOTA pages uses the archive name as well.
```suggestion
application, the metadata will be automatically parsed from the .elf and .config file. The parsed data is stored in `ble_mesh_metadata.json` and put into a `dfu_application.zip` archive in the build folder of the application. Additionally, the
```

File: doc/nrf/protocols/bt/bt_mesh/dfu_over_bt_mesh.rst [hunk 3]
```diff
@@ -182,8 +182,31 @@ To bring a stalled Target node to idle state, use the ``mesh models dfu cli canc
...
 
-Composing the firmware metadata
-*******************************
+Generating the firmware metadata
+********************************
+
+Automated metadata generation
+=============================
+
+The most efficient way to generate the required DFU metadata is to use the DFU metadata extraction script integrated in the NCS build system. By enabling the ``CONFIG_BT_MESH_DFU_METADATA_ON_BUILD`` option in the
```
Comment (PavelVPV) [inline doc/nrf/protocols/bt/bt_mesh/dfu_over_bt_mesh.rst:191]: Every new sentence should be on a new line and there is no limit on the line length (ncs requirement, it is opposite in zephyr).

File: doc/nrf/protocols/bt/bt_mesh/dfu_over_bt_mesh.rst [hunk 4]
```diff
@@ -182,8 +182,31 @@ To bring a stalled Target node to idle state, use the ``mesh models dfu cli canc
...
+application, the metadata will be automatically parsed from the .elf and .config file. The parsed data is put into a DFU zip-archive in the build folder of the application in a JSON format. Additionally, the
+metadata string, required by the ``mesh models dfu metadata encode`` command, will be printed in the command line window when the application is built.
```
Comment (PavelVPV) [inline doc/nrf/protocols/bt/bt_mesh/dfu_over_bt_mesh.rst:193]: > required by the ``mesh models dfu metadata encode``

It is required by `dfu slot add`:
https://github.com/nrfconnect/sdk-nrf/blob/b962460658b241014e4c6274e03cb52f2b1d5ed6/doc/nrf/protocols/bt/bt_mesh/dfu_over_bt_mesh.rst?plain=1#L52-L58

File: doc/nrf/protocols/bt/bt_mesh/dfu_over_bt_mesh.rst [hunk 5]
```diff
@@ -182,8 +182,31 @@ To bring a stalled Target node to idle state, use the ``mesh models dfu cli canc
...
+
+.. note::
+   Currently this script only supports applications that contain a single composition data structure, meaning that applications containing multiple composition data structures must rely on manual generation of
+   the metadata string(s). Additionally, the script is hardcoded to produce a metadata string where the firmware is targeted for the application core.
```
Comment (PavelVPV) [inline doc/nrf/protocols/bt/bt_mesh/dfu_over_bt_mesh.rst:197]:
```suggestion
.. note::
   Currently this script only supports applications that contain a single composition data structure, meaning that applications containing multiple composition data structures must rely on manual generation of the metadata string(s).
   It is also required that the composition data is declared with const-qualifier.
   Additionally, the script is hardcoded to produce a metadata string where the firmware is targeted for the application core.
```

**Review by PavelVPV — CHANGES_REQUESTED** [review]

File: doc/nrf/protocols/bt/bt_mesh/dfu_over_bt_mesh.rst (hunk 4 above)
Comment (mia-ko) [inline doc/nrf/protocols/bt/bt_mesh/dfu_over_bt_mesh.rst:193]:
```suggestion
By enabling the :kconfig:option:`CONFIG_BT_MESH_DFU_METADATA_ON_BUILD` option in the application, the metadata will be automatically parsed from the ``.elf`` and ``.config`` files.
The parsed data is stored in the :file:`ble_mesh_metadata.json` file.
The file is placed in the :file:`dfu_application.zip` archive in the build folder of the application.
Additionally, the metadata string required by the ``mesh models dfu slot add`` command will be printed in the command line window when the application is built.
```

File: doc/nrf/protocols/bt/bt_mesh/dfu_over_bt_mesh.rst [hunk 6]
```diff
@@ -182,8 +182,31 @@ To bring a stalled Target node to idle state, use the ``mesh models dfu cli canc
...
+
+In addition to printing the metadata string on build, a separate west command can be utilized to print the metadata to the console, given that it is already generated by the build system. This provides the user
+easy access to this information without having to enter the .json file in the build folder or to rebuild the application::
```
Comment (mia-ko) [inline doc/nrf/protocols/bt/bt_mesh/dfu_over_bt_mesh.rst:200]:
```suggestion

In addition to printing the metadata string when the application is built, a separate west command can be utilized to print the metadata to the console, given that it is already generated by the build system.
This gives the user easy access to this information, without having to enter the .json file in the build folder or to rebuild the application::
```

File: doc/nrf/protocols/bt/bt_mesh/dfu_over_bt_mesh.rst [hunk 7]
```diff
@@ -182,8 +182,31 @@ To bring a stalled Target node to idle state, use the ``mesh models dfu cli canc
...
+
+  west ble-mesh-dfu-metadata
+
+For this particular example, the output generated by either the build system or the west ble-mesh-dfu-metadata command would be::
```
Comment (mia-ko) [inline doc/nrf/protocols/bt/bt_mesh/dfu_over_bt_mesh.rst:204]:
```suggestion
For this particular example, the output generated by either the build system or the west ``ble-mesh-dfu-metadata`` command would be::
```

**Review by mia-ko — CHANGES_REQUESTED** [review]

File: doc/nrf/protocols/bt/bt_mesh/dfu_over_bt_mesh.rst (hunk 4 above)
Comment (Andrewpini) [reply doc/nrf/protocols/bt/bt_mesh/dfu_over_bt_mesh.rst:193]: Right... That is what I meant to write 👍

File: doc/nrf/protocols/bt/bt_mesh/dfu_over_bt_mesh.rst (hunk 2 above)
Comment (Andrewpini) [reply doc/nrf/protocols/bt/bt_mesh/dfu_over_bt_mesh.rst:192]: @Balaklaka While you are correct, I think it just becomes more confusing if we add `dfu_application.zip_ble_mesh_metadata.json`. I think it is easier if the user just have to focus on what is inside the zip.

File: doc/nrf/protocols/bt/bt_mesh/dfu_over_bt_mesh.rst [hunk 8]
```diff
@@ -182,8 +182,43 @@ To bring a stalled Target node to idle state, use the ``mesh models dfu cli canc
...
+
+  Bluetooth Mesh Composition metadata generated:
+    Encoded metadata: 020000000100000094cf24017c26f3710100
+    Full metadata written to: APPLICATION_FOLDER\build\zephyr\dfu_application.zip
+
+.. note::
+   Currently this script only supports applications that contain a single composition data structure, meaning that applications containing multiple composition data structures must rely on manual generation of the metadata string(s).
+   Additionally, the script is hardcoded to produce a metadata string where the firmware is targeted for the application core.
+
+In addition to printing the metadata string when the application is built, a separate west command can be utilized to print the metadata to the console, given that it is already generated by the build system.
```
Comment (PavelVPV) [inline doc/nrf/protocols/bt/bt_mesh/dfu_over_bt_mesh.rst:211]: The last part of the sentence tells the same as the first one. I'd suggest to remove the first part of the sentence:
```suggestion
A separate west command can be utilized to print the metadata to the console, given that it is already generated by the build system.
```

File: doc/nrf/protocols/bt/bt_mesh/dfu_over_bt_mesh.rst (hunk 5 above)
Comment (PavelVPV) [reply doc/nrf/protocols/bt/bt_mesh/dfu_over_bt_mesh.rst]: This part is missing:
> It is also required that the composition data is declared with const-qualifier.

File: doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst [hunk 9]
```diff
@@ -76,6 +76,10 @@ Bluetooth® LE
 Bluetooth Mesh
 --------------
 
+* Added:
+ * Script for extracting Mesh DFU metadata automatically on building application, enabled by the :kconfig:option:`CONFIG_BT_MESH_DFU_METADATA_ON_BUILD` Kconfig option.
+ * Separate command ``west build -t ble_mesh_dfu_metadata`` to print already generated Mesh DFU metadata to terminal.
+
```
Comment (mia-ko) [inline doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst:82]:
```suggestion
* Added:
  
  * Script for extracting the Bluetooth Mesh DFU metadata automatically when building an application.
    The script is enabled by the :kconfig:option:`CONFIG_BT_MESH_DFU_METADATA_ON_BUILD` Kconfig option.
  * Separate command ``west build -t ble_mesh_dfu_metadata`` to print already generated Bluetooth Mesh DFU metadata to a terminal.

```

**Review by mia-ko — CHANGES_REQUESTED** [review]

**Review by PavelVPV — APPROVED** [review]

**Review by mia-ko — APPROVED** [review]

**Pull request merged**